commands/command_pre_push.go

```diff
@@ -137,7 +133,19 @@ func prePushRef(left, right string) {
 			os.Exit(2)
 		}
 	}
+}
+
+func filteredPointers(pointers []*lfs.WrappedPointer, filter lfs.StringSet) []*lfs.WrappedPointer {
```

Comment:
BTW just wanted to acknowledge you're absolutely right about the concurrency thing. Temporary brain fart.

I'm working on an update that attempts to centralize the uploading logic across pre-push and push. I'll probably post that as a PR against this PR so we can review that, and discuss package options there.